Added entry for Best-First Search (#1876)
Codecademy/docs: content/ai/concepts/search algorithms/best first search.md

---
Title: 'Best-First Search'

> Description: 'A graph searching algorithm that expands a node based on an evaluation function.'

Reviewer: ```suggestion
Description: 'A graph searching algorithm that determines a path between nodes based on an evaluation function.'
```

Reviewer: I just found "expands a node" very abstract and opaque. I've provided an alternative, but feel free to try something else.

Subjects:
  - 'Data Science'
  - 'AI'
Tags:
  - 'Search Algorithms'
  - 'AI'
CatalogContent:
  - 'learn-python-3'
  - 'paths/machine-learning-ai-engineering-foundations'
---

> **Best-First Search** is a class of search algorithms that searches a graph (the graph can be weighted or unweighted) for a path from some start node to a goal node. These algorithms are known as informed search algorithms, meaning that they incorporate information regarding the location of the goal node relative to any other node. This information is quantified by an evaluation function denoted by _f(x)_. In a weighted graph, the evaluation function can combine information regarding the edge weights and some heuristic function, _h(x)_, that represents a measure from a node (_x_ in this case) to the goal node. When searching for the goal node, a BFS algorithm (not to be confused with Breadth-First Search) selects the node with the most promising value determined by the evaluation function.

Reviewer: ```suggestion
**Best-first search** is a class of search algorithms that explores a graph (the graph can be weighted or unweighted) for a path from some start node to a goal node. These algorithms are known as informed search algorithms, meaning that they incorporate information regarding the location of the goal node relative to any other node. This information is quantified by an evaluation function denoted by _f(x)_. In a weighted graph, the evaluation function can combine information regarding the edge weights and some heuristic function, _h(x)_, that represents a measure from a node (_x_ in this case) to the goal node. When searching for the goal node, a BFS algorithm (not to be confused with breadth-first search) selects the node with the most promising value determined by the evaluation function.
```

In a weighted graph, the total cost to go from the start node to the goal node is the sum of the cost of every visted edge along the path. The values from the evaluation function do not contribute to the total cost. They are only there to guide the search.

## Best-First Search Algorithms
The following is a list of Best-First search algorithms.

> - Greedy Best-First Search

Reviewer 2: Can we have a one line summary for each of these?

Author: Done! :grin:

> ## Best-First Search Algorithms

Reviewer: ```suggestion
## Best-First Search Algorithms

```

> - Greedy Best-First Search: A search algorithms where the evaluation function is strictly equal to the heuristic function, disregarding the edge weights in a weighted graph.

Reviewer: ```suggestion
- Greedy Best-First Search: A search algorithm where the evaluation function is strictly equal to the heuristic function, disregarding the edge weights in a weighted graph.
```
